ui/src/app/pages/analysis/notebook-redirect.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+import {ClusterApiStub} from 'testing/stubs/cluster-api-stub';
+import {JupyterApiStub} from 'testing/stubs/jupyter-api-stub';
+import {NotebooksApiStub} from 'testing/stubs/notebooks-api-stub';
+import {NotebooksClusterApiStub} from 'testing/stubs/notebooks-cluster-api-stub';
+import {ProfileStubVariables} from 'testing/stubs/profile-api-stub';
+import {workspaceStubs, WorkspaceStubVariables} from 'testing/stubs/workspaces-api-stub';
+import {NotebookRedirect} from './notebook-redirect';
+
+import {exportFunctions} from 'app/utils/index';
+
+describe('NotebookRedirect', () => {
+  const workspace = {
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.OWNER,
+  };
+  const profile = ProfileStubVariables.PROFILE_STUB;
+  const reload = jest.fn();
+  const updateCache = jest.fn();
+
+  let clusterStub: ClusterApiStub;
+
+  /*
+  NotebookRedirect.progressCardStates:
+  {includes: [Progress.Unknown, Progress.Initializing, Progress.Resuming], icon: 'notebook'},
+  {includes: [Progress.Authenticating], icon: 'success-standard'},
+  {includes: [Progress.Creating, Progress.Copying], icon: 'copy'},
+  {includes: [Progress.Redirecting], icon: 'circle-arrow'}
+   */
+  const progressCardIds = {
+    unknownInitializingResuming: 0,
+    authenticating: 1,
+    creatingCopying: 2,
+    redirecting: 3
+  };
+
+  const component = () => {
+    return mount(<NotebookRedirect/>);
+  };
+
+  const getCardSpinnerTestId = (cardId) => {
+    return '[data-test-id="progress-card-spinner-' + cardId + '"]';
+  };
+
+  beforeEach(() => {
+    clusterStub = new ClusterApiStub();
+    clusterStub.cluster.status = ClusterStatus.Creating;
+
+    registerApiClient(ClusterApi, clusterStub);
+    registerApiClientNotebooks(JupyterApi, new JupyterApiStub());
+    registerApiClientNotebooks(NotebooksApi, new NotebooksApiStub());
+    registerApiClientNotebooks(NotebooksClusterApi, new NotebooksClusterApiStub());
+
+    serverConfigStore.next({gsuiteDomain: 'x'});
+    urlParamsStore.next({
+      ns: WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
+      wsid: WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
+      nbName: 'blah blah'
+    });
+    queryParamsStore.next({
+      kernelType: Kernels.R,
+      creating: true
+    });
+    currentWorkspaceStore.next(workspace);
+    userProfileStore.next({profile, reload, updateCache});
+
+    // mock timers
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    jest.clearAllMocks();
+  });
+
+  it('should render', () => {
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+  });
+
+  it('should show redirect display before showing notebook', async() => {
+    const wrapper = component();
+    expect(wrapper.exists('[data-test-id="notebook-redirect"]')).toBeTruthy();
+  });
+
+  it('redirect state for existing notebook should be "Initializing" until cluster is running', async() => {
```

Comment:
nit: it's nice if this can read like a sentence, like `it('puts existing notebook in redirect state')`

Reply:
updated